### ISSUE TITLE
Running large amounts of jobs gives some (harmless?) errors

### ISSUE THREAD

**Issue by jonasc** [pgiri/dispy#47]
I'm currently using dispy for a huge number of tasks which tends to give me some errors which appear not reproducibly.
I have run the following code:
~~~python
#!/usr/bin/env python3.5

import dispy
import dispy.httpd
import logging


def results(job):
    if job.status == dispy.DispyJob.Terminated:
        print('ERROR finishing job "{}"'.format(job.id))
    del jobs[job.id]


def dispy_test():
    import socket
    host = socket.gethostname()
    return host

if __name__ == '__main__':
    cluster = dispy.JobCluster(dispy_test, nodes=['bremen.imp.fu-berlin.de', 'cmp135.imp.fu-berlin.de'], loglevel=logging.DEBUG, ext_ip_addr='87.77.51.134', callback=results)
    jobs = dict()
    for i in range(10000):
        job = cluster.submit()
        job.id = i
        jobs[job.id] = job
    cluster.wait()
    cluster.print_status()
    print('missing jobs:', jobs)
    cluster.close()
~~~
Running this typically yields a handful of the following errors:
~~~python
Traceback (most recent call last):
  File "XXX/lib/python3.5/site-packages/asyncoro/__init__.py", line 3706, in _schedule
    retval = coro._generator.send(coro._value)
  File "YYY/dispy/py3/dispy/__init__.py", line 1629, in run_job
    _job.job.id, _job.uid, node.ip_addr, node.busy, node.cpus)
AttributeError: 'NoneType' object has no attribute 'id'
~~~
As the output indicates there is no job missing and in the end they are all removed from the dict.
~~~
Total job time: 64.503 sec, wall time: 36.580 sec, speedup: 1.763

missing jobs: {}
~~~

So it looks like there should just be an try/except surrounding the particular code?

**Comment by pgiri** [comment]
It seems a job is finished before 'run_job' executes 'else' part (as in your example, job takes very little time). Fix for this committed with b13b593.

Yes, that exception is harmless; unlike with threads, with asyncoro a coroutine raising an exception is harmless to rest (while that coroutine is terminated due to exception).

**Comment by pgiri** [comment]
This has been addressed in 4.6.15 release.
